python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -315,7 +316,9 @@ def execute_asset_backfill_iteration_inner(
     recently_materialized_asset_partitions = AssetGraphSubset(asset_graph)
     for asset_key in asset_backfill_data.target_subset.asset_keys:
         records = instance_queryer.get_materialization_records(
-            asset_key=asset_key, after_cursor=asset_backfill_data.latest_storage_id
+            asset_key=asset_key,
+            after_cursor=asset_backfill_data.latest_storage_id,
+            tags=frozendict({BACKFILL_ID_TAG: backfill_id}),
```

Comment:
Depends on https://github.com/dagster-io/dagster/pull/11509 to work in the unmigrated case